File: domain/src/main/java/io/prizy/domain/contest/mapper/UpdateContestMapper.java
package io.prizy.domain.contest.mapper;


import io.prizy.domain.contest.command.UpdateContest;
import io.prizy.domain.contest.model.Contest;
import lombok.experimental.UtilityClass;

@UtilityClass
public class UpdateContestMapper {

  public Contest map(UpdateContest create) {
    return Contest.builder()
      .id(create.id())
      .name(create.name())
      .description(create.description())
      .category(create.category())
      .fromDate(create.fromDate())
      .toDate(create.toDate())
      .assetIds(create.assetIds())
      .coverAssetId(create.coverAssetId())
      .packs(create.packs().stream().map(CreatePackMapper::map).toList())
      .cost(create.cost())
      .facebookPage(create.facebookPage())
      .instagramPage(create.instagramPage())
      .website(create.website())
      .newsletterSubscription(create.newsletterSubscription())
      .adultOnly(create.adultOnly())
      .merchant(create.merchant())
      .boardId(create.boardId())
      .build();
  }

}

File: public-api/src/main/kotlin/io/prizy/publicapi/port/contest/mapper/UpdateContestMapper.kt
package io.prizy.publicapi.port.contest.mapper

import io.prizy.domain.contest.model.ContestCategory
import io.prizy.domain.contest.model.UpdateContest
import io.prizy.publicapi.port.contest.graphql.dto.ContestCategoryDto
import io.prizy.publicapi.port.contest.graphql.dto.UpdateContestDto
import java.util.Optional

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 8:47 PM
 */

object UpdateContestMapper {

  fun map(dto: UpdateContestDto): UpdateContest = UpdateContest.builder()
    .id(dto.id)
    .name(dto.name)
    .description(dto.description)
    .category(map(dto.category))
    .fromDate(dto.fromDate)
    .toDate(dto.toDate)
    .assetIds(dto.assetIds)
    .coverAssetId(dto.coverAssetId)
    .packs(dto.packs.map(CreatePackDtoMapper::map))
    .cost(dto.cost)
    .facebookPage(Optional.ofNullable(dto.facebookPage))
    .instagramPage(Optional.ofNullable(dto.instagramPage))
    .website(Optional.ofNullable(dto.website))
    .newsletterSubscription(dto.newsletterSubscription)
    .adultOnly(dto.adultOnly)
    .merchant(CreateMerchantDtoMapper.map(dto.merchant))
    .boardId(dto.boardId)
    .build()

  fun map(dto: ContestCategoryDto): ContestCategory = ContestCategory.valueOf(dto.name)
}
